src/Maybe.ts

```diff
@@ -0,0 +1,15 @@
+export class Maybe<T> {
+    private value;
+
+    constructor(value: T) {
+        this.value = value;
+    }
+
+    public map(fn: Function): Maybe<T> {
+        return new Maybe(this.value !== null && this.value !== undefined ? fn(this.value) : null);
+    }
+
+    public getOrDefault(defaultValue: any): any {
```

Comment:
Default value should be T

Reply:
Nope, I can return any defaultValue, for example:
If T is string or number, I can have defaultValue = null
No?

Reply:
Did you explore the possibility of nullable type as generic arg?